### PR TITLE
core: refactor webhook config

### LINKED ISSUE #10025: refactor webhook creation
refactor webhook creation using helper funtion

### PR DESCRIPTION
**Pull request by subhamkrai**
**Description of your changes:**

refactor validating webhook config.

Closes: https://github.com/rook/rook/issues/10025
Signed-off-by: subhamkrai <srai@redhat.com>

**Which issue is resolved by this Pull Request:**
Resolves #10025 

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

File: pkg/operator/ceph/webhook-config.go [hunk 1]
```diff
@@ -166,11 +169,15 @@ func fetchValidatingWebhookConfig(ctx context.Context, clusterdContext *clusterd
 }
 
 func createValidatingWebhookConfig(ctx context.Context, clusterdContext *clusterd.Context) error {
-	sideEffects := admv1.SideEffectClassNone
-	var timeout int32 = 5
-	serviceCephClusterPath := "/validate-ceph-rook-io-v1-cephcluster"
-	serviceCephBlockPoolPath := "/validate-ceph-rook-io-v1-cephblockpool"
-	serviceCephObjectStorePath := "/validate-ceph-rook-io-v1-cephobjectstore"
+	name := []string{
```
Comment (leseb) [inline pkg/operator/ceph/webhook-config.go:172]: Give the variable a better name?

Comment (subhamkrai) [reply pkg/operator/ceph/webhook-config.go]: modified, naming is always a challenge :slightly_smiling_face:

File: pkg/operator/ceph/webhook-config.go [hunk 2]
```diff
@@ -182,91 +189,14 @@ func createValidatingWebhookConfig(ctx context.Context, clusterdContext *cluster
...
-				TimeoutSeconds:          &timeout,
-			},
-		},
 	}
 
+	for i := range resources {
+		validatingWebhook.Webhooks = append(validatingWebhook.Webhooks, webhookRules(resourcesWebhookName[i], resources[i], resourcesServicePath[i]))
+	}
+
+	validatingWebhook.Webhooks = append(validatingWebhook.Webhooks, webhookRules(fmt.Sprintf("cephcluster-wh-%s-%s.rook.io", admissionControllerAppName, namespace), "", ""))
```
Comment (travisn) [inline pkg/operator/ceph/webhook-config.go:198]: Is this a duplicate from the other entry for cephcluster in the slice above, or what is different about it? And why not include it in the slice above?

Comment (subhamkrai) [reply pkg/operator/ceph/webhook-config.go]: actually it should be inside the loop, leftover from testing

**Review by travisn — APPROVED** [review]

**Review by leseb — APPROVED** [review]

**Pull request merged**